onnxruntime/core/providers/webnn/builders/model_builder.cc

```diff
@@ -248,10 +264,21 @@ Status ModelBuilder::AddOperandFromPersistMemoryBuffer(
   auto persist_buffer = std::make_unique<uint8_t[]>(size);
   uint8_t* dest = persist_buffer.get();
   memcpy(dest, buffer, size);
-  emscripten::val view{emscripten::typed_memory_view(size / element_size, reinterpret_cast<const float*>(dest))};
+  emscripten::val view = emscripten::val::undefined();
   emscripten::val desc = emscripten::val::object();
+  if (element_size == 2) {
```

Comment:
It's better to use data type as variable to allocate different types of buffer view.